src/flags.ml

```diff
@@ -2974,8 +2976,12 @@ module Global = struct
         enable_values
         (string_of_enable enable_default_after)
     )
+    
   (* let enable mdl = enabled := mdl :: !enabled *)
-  let enabled () = !enabled
+  let enabled () = 
+      if (List.mem `BMCREACHABLE !enabled |> not) && (List.mem `BMC !enabled)
+      then `BMCREACHABLE :: !enabled
+      else !enabled
```

Comment:
Move this check to `kind2Flow` so that the `BMCREACHABLE` module is only added if there is at least one reachability property with a lower bound.